src/App.js

```diff
@@ -3,20 +3,28 @@ import React from "react";
 import AddTodoForm from "./AddTodoForm";
 import TodoList from "./TodoList";
 
-const useSemiPersistentState = () => {
-  const [todoList, setTodoList] = React.useState(
-    JSON.parse(localStorage.getItem("savedTodoList"))
-  );
+function App() {
+  const [todoList, setTodoList] = React.useState([]);
+  const [isLoading, setIsLoading] = React.useState(true);
 
   useEffect(() => {
-    localStorage.setItem("savedTodoList", JSON.stringify(todoList));
-  }, [todoList]);
-
-  return [todoList, setTodoList];
-};
+    new Promise((resolve, reject) =>
+      setTimeout(() => {
+        resolve({
+          data: { todoList: JSON.parse(localStorage.getItem("savedTodoList")) },
+        });
+      }, 2000)
+    ).then((result) => {
+      setTodoList(result.data.todoList);
+      setIsLoading(false);
+    });
+  }, []);
 
-function App() {
-  const [todoList, setTodoList] = useSemiPersistentState();
+  useEffect(() => {
+    if (isLoading !== true) {
```

Comment:
Nice job here!! 💯 
But to make the code more terse you do something like this: `!isLoading`.
So you're is not loading. Does that make sense?